core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchResultsImpl.java

```diff
@@ -62,85 +76,149 @@
-    configuredResultsPerPage = getConfiguredResultsPerPage();
-    Optional.ofNullable(request.adaptTo(RequestWrapper.class))
-        .ifPresent(requestWrapper -> {
-          i18n = requestWrapper.getI18n();
-          requestWrapper.getParameter(SEARCH_TERM).ifPresent(s -> term = s);
-          startPage = requestWrapper.getParameter(QUERY_PARAM_START)
-              .map(s -> new StringToInteger(s).asInt())
-              .map(OptionalInt::getAsInt)
-              .orElse(DEFAULT_START_PAGE);
-          resultsPerPage = resolveResultsPerPage(requestWrapper);
-          //todo: remove this when ICSAAS-315 is done - currently utilized only for demo purpose
-          loadMoreRows = resultsPerPage + configuredResultsPerPage;
-          SearchConfiguration searchConfiguration = request.getResource().adaptTo(ConfigurationBuilder.class)
-              .as(SearchConfiguration.class);
-          FulltextSearchGetRequestPayload fulltextSearchGetRequestPayload =
-              DefaultFulltextSearchRequestPayload.builder(new DefaultTermQuery(term),
-                      new DefaultLanguageQuery(getLanguage()))
-                  .optionalQuery(
-                      new PaginationQuery(startPage, resultsPerPage,
-                          fulltextSearchConfigurationService.getRowsMaxLimit()))
-                  .optionalQuery(new HighlightingTagQuery(Highlighting.HIGHLIGHTING_TAG_NAME))
-                  .optionalQuery(FiltersQuery.builder()
-                      .filterEntries(Arrays.stream(searchConfiguration.searchFilters()).collect(Collectors.toMap(
-                          SearchFilterConfiguration::name, SearchFilterConfiguration::value))).build())
-                  .build();
-          results = fulltextSearchService.getResults(searchConfiguration.index(), fulltextSearchGetRequestPayload)
-              .map(FulltextSearchResults::getResults).orElse(
-                  Collections.emptyList());
-        });
+    Optional<RequestWrapper> requestWrapper = getRequestWrapper();
+    requestWrapper
+        .flatMap(this::getSearchTerm)
+        .ifPresent(searchTerm -> initSearch(searchTerm, requestWrapper.get()));
+  }
+
+  @JsonIgnore
+  @Override
+  public Set<Filter> getFilters() {
+    return mergedFilters;
+  }
+
+  private Optional<RequestWrapper> getRequestWrapper() {
+    return Optional.ofNullable(request.adaptTo(RequestWrapper.class));
+  }
+
+  private Optional<String> getSearchTerm(RequestWrapper requestWrapper) {
+    return requestWrapper.getParameter(SEARCH_TERM).filter(StringUtils::isNotBlank);
   }
 
-  private int resolveResultsPerPage(RequestWrapper requestWrapper) {
+  private int getStartPage(RequestWrapper requestWrapper) {
+    return requestWrapper.getParameter(QUERY_PARAM_START)
+        .map(start -> new StringToInteger(start).asInt())
+        .map(OptionalInt::getAsInt)
+        .orElse(DEFAULT_START_PAGE);
+  }
+
+  private void initSearch(String searchTerm, RequestWrapper requestWrapper) {
```

Comment:
that method is pretty big still. can  you introduce few wrapper functions, e.g. for initializing fulltextSearchGetRequestPayload  ?